File: test_dir/cpp/src/lists/as_arg.cpp
#include "lists\as_arg.h"

void list_as_arg(const PyList<int> &a) {
    PyList<int> b = a * 2;
    print(b);
}
void list_as_mutable_arg(PyList<int> &a) {
    a *= 3;
    print(a);
}

File: test_dir/python/src/lists/as_arg.py
from test_dir.python.pypp.custom_types import PyppMut


def list_as_arg(a: list[int]):
    b: list[int] = a * 2
    print(b)


def list_as_mutable_arg(a: PyppMut(list[int])):
    a *= 3
    print(a)
